dapp/src/components/page/governance/ProposalForm.tsx

```diff
@@ -0,0 +1,326 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import {
+  MDXEditor,
+  headingsPlugin,
+  listsPlugin,
+  quotePlugin,
+  thematicBreakPlugin,
+  markdownShortcutPlugin,
+  linkPlugin,
+  linkDialogPlugin,
+  imagePlugin,
+  tablePlugin,
+  codeBlockPlugin,
+  codeMirrorPlugin,
+  diffSourcePlugin,
+  frontmatterPlugin,
+  toolbarPlugin,
+  Separator,
+  UndoRedo,
+  BoldItalicUnderlineToggles,
+  BlockTypeSelect,
+  CodeToggle,
+  CreateLink,
+  DiffSourceToggleWrapper,
+  InsertAdmonition,
+  InsertImage,
+  InsertTable,
+  InsertFrontmatter,
+  InsertThematicBreak,
+  InsertCodeBlock,
+  ListsToggle,
+  type CodeBlockEditorDescriptor,
+  useCodeBlockEditorContext,
+} from "@mdxeditor/editor";
+import "@mdxeditor/editor/style.css";
+import * as Delegation from "@web3-storage/w3up-client/delegation";
+import * as Client from "@web3-storage/w3up-client";
+import { capitalizeFirstLetter, getIpfsBasicLink } from "utils/utils";
+import type { ProposalOutcome } from "types/proposal";
+import Loading from "components/utils/Loading";
+
+const ProposalForm: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [mdText, setMdText] = useState("");
+  const [isClient, setIsClient] = useState(false);
+  const [approveDescription, setApproveDescription] = useState("");
+  const [rejectDescription, setRejectDescription] = useState("");
+  const [cancelledDescription, setCancelledDescription] = useState("");
+  const [approveXdr, setApproveXdr] = useState("");
+  const [rejectXdr, setRejectXdr] = useState("");
+  const [cancelledXdr, setCancelledXdr] = useState("");
+  const [imageFiles, setImageFiles] = useState<
+    { localUrl: string; publicUrl: string; source: File }[]
+  >([]);
+
+  useEffect(() => {
+    setIsClient(true);
+  }, []);
+
+  if (!isClient) {
+    return null;
+  }
+
+  const imageUploadHandler = async (image: File) => {
+    const url = URL.createObjectURL(image);
+    setImageFiles([
+      ...imageFiles,
+      { localUrl: url, publicUrl: image.name, source: image },
+    ]);
+    return url;
+  };
+
+  const PlainTextCodeEditorDescriptor: CodeBlockEditorDescriptor = {
+    match: (_language, _meta) => true,
+    priority: 0,
+    Editor: (props) => {
+      const cb = useCodeBlockEditorContext();
+      return (
+        <div onKeyDown={(e) => e.nativeEvent.stopImmediatePropagation()}>
+          <textarea
+            rows={5}
+            cols={20}
+            defaultValue={props.code}
+            onChange={(e) => cb.setCode(e.target.value)}
+          />
+        </div>
+      );
+    },
+  };
+
+  const submitProposal = async () => {
```

Comment:
Where is the maintainer check? We only want maintainers to be able to submit proposals.